src/meta-srv/src/procedure/alter_table.rs

```diff
@@ -390,8 +392,8 @@ impl Procedure for AlterTableProcedure {
 
         match self.data.state {
             AlterTableState::UpdateMetadata => self.on_update_metadata().await,
-            AlterTableState::DatanodeAlterTable(_) => self.on_datanode_alter_table().await,
-            AlterTableState::Broadcast => self.on_broadcast().await,
+            AlterTableState::InvalidateTableCache => self.on_broadcast().await,
+            AlterTableState::DatanodeAlterTable => self.on_datanode_alter_table().await,
```

Comment:
Imagine the following scenario:
1. invalidate cache 
2. processing alter table
3. users request this table
4. get metadata from metasrv and put it into cache
5. alter table complete
6. dirty data in cache?

Reply:
It updates the metadata at first.